Real supervision

benlangfeld opened the pull request:
@bklang ?

bklang commented on LICENSE.txt:
```diff
@@ -1,4 +1,4 @@
-Copyright (c) 2011 Ben Langfeld, Jay Phillips
+Copyright (c) 2016 Ben Langfeld, Jay Phillips
```
2011-2016

bklang commented on CHANGELOG.md:
```diff
@@ -1,4 +1,5 @@
 # [develop](https://github.com/adhearsion/ruby_ami)
+  * Breaking change: Removed the deprecated `RubyAMI::Client` because it is no longer relevant.
```
Doesn't this conflict with the other PR?

benlangfeld replied:
It's just that one is based on the other.

bklang commented:
The title of this PR is "Real supervision" but I only see one relevant code change: `async.run`. Is that really all that is required to have it properly supervised?

benlangfeld commented:
> Is that really all that is required to have it properly supervised?

Not for it to *be* supervised, but for it to fit into Celluloid supervision semantics. I added a documentation note to clarify. I also corrected an omission which was updated in the docs but not implemented; I submitted this PR in the midst of production downtime and forgot about it.

bklang commented:
:+1:
